lib/pavilion/plugins/commands/result.py

```diff
@@ -278,7 +292,16 @@ def update_results(self, pav_cfg: dict, tests: List[TestRun],
                 return False
 
             # The new results will be attached to the test (but not saved).
-            test.gather_results(test.results.get('return_value', 1),
-                                regather=True, log_file=log_file)
+            results = test.gather_results(test.results.get('return_value', 1),
+                                          regather=True, log_file=log_file)
+
+            if save:
+                results_log = test.path/'results.log'
+                test.save_results(results)
+                with results_log.open('a') as log:
```

Comment:
You don't need to reopen the log file and append to it; because if you're saving `log_file` should already be that file, open.  Also, I'm pretty sure there's a 'result_log_path' variable on the test object already.

Reply:
It looks like the log_file that is passed to this method is used for the command argument `--show-log`, and is either None or a StringIO. It isn't a test object's log file. Furthermore, the update_results method takes a list of Test_Run objects and iterates over them, because of this I have to open the results_log for each test instance...